### ISSUE TITLE
Use TypeScript

### ISSUE THREAD

**Issue by openjck** [untitled-data-project/glitter#2]
I don't actually have a strong opinion on this at all, but if we're going to do it, it would be easier to do it sooner rather than later.

Should we use TypeScript? (Feel free to close as a way of answering, "No.")

**Comment by spasovski** [comment]
Definitely against this - as JS progresses and best practices change it's good to know why something was intentionally written a certain way and compilers tend to abstract that away. I say this knowing we use svelte.

**Comment by hamilton** [comment]
I would prefer to use TypeScript if possible, personally, but we can't quite yet with Svelte since it's not supported yet.

I think we're probably a good ways away from types making their way into JS natively (nothing in any of the [TC39 proposals](https://github.com/tc39/proposals), and in the meantime we have code to write. TS would make it all substantially cleaner and much more solid. This all said, typescript really doesn't transform the language as much as you might think.

**Comment by openjck** [comment]
I think [Svelte does support it now](https://github.com/dafn/svelte-typescript-rollup) but again, I really have no opinion one way or the other, except that all of this configuration-type stuff (ESLint, Prettier, TypeScript) is easier to start with than to retrofit later.

**Comment by hamilton** [comment]
Yep, there are a few TS preprocessors for rollup / svelte (transpiling the script tags), but not actual TS support (type checking on components themselves) so you lose the good parts at component boundaries. [This issue](https://github.com/sveltejs/svelte/issues/3677) is part of a long chain of people asking for it over 2019.

That doesn't preclude us from using a TS preprocessor, but it doesn't really give it quite as much value.

**Comment by mistlog** [comment]
> Yep, there are a few TS preprocessors for rollup / svelte (transpiling the script tags), but not actual TS support (type checking on components themselves) so you lose the good parts at component boundaries. [This issue](https://github.com/sveltejs/svelte/issues/3677) is part of a long chain of people asking for it over 2019.
> 
> That doesn't preclude us from using a TS preprocessor, but it doesn't really give it quite as much value. This doesn't rule out using TS today in the hope that it'll provide more benefit later. We should probably deliberate this over zoom.

Just at the bottom of this issue, I proposed another approach to support typescript completely, it's working in progress now, however, there is no response from svelte and there is no user feedback, it may be hard for me to continue on, please give me some suggestions if it's helpful for you.
